### PR TITLE
rec: add structured logging backend that uses JSON representation

### PR DESCRIPTION
**Pull request by omoerbeek**
### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [X] included documentation (including possible behaviour changes)
- [X] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/docs/appendices/structuredlogging.rst [hunk 1]
```diff
@@ -100,3 +101,13 @@ To query the log, use a command similar to::
 
   # journalctl -r -n 1 -o json-pretty -u pdns-recursor.service
 
+The ``json`` backend
+--------------------
+The ``json`` structured logging backend has been added in version 5.1.0 and uses the same keys and values as the default backend.
+And example of a a log object::
```
Comment (fredmorcos) [inline pdns/recursordist/docs/appendices/structuredlogging.rst:107]:
```suggestion
An example of a log object:
```

File: pdns/recursordist/rec-main.cc [hunk 2]
```diff
@@ -2986,6 +3030,31 @@ static void handleRuntimeDefaults(Logr::log_t log)
...
+{
+  if (logname == "systemd-journal") {
+#ifdef HAVE_SYSTEMD
+    if (int fileDesc = sd_journal_stream_fd("pdns-recusor", LOG_DEBUG, 0); fileDesc >= 0) {
+      g_slog = Logging::Logger::create(loggerSDBackend);
+      close(fileDesc);
+    }
+#endif
+    if (g_slog == nullptr) {
+      cerr << "Structured logging to systemd-journal requested but it is not available" << endl;
```
Comment (fredmorcos) [inline pdns/recursordist/rec-main.cc:3043]:
```suggestion
      cerr << "Requested structured logging to systemd-journal, but it is not available" << endl;
```

File: pdns/recursordist/rec-main.cc [hunk 3]
```diff
@@ -2986,6 +3030,31 @@ static void handleRuntimeDefaults(Logr::log_t log)
...
+    }
+  }
+  else if (logname == "json") {
+    g_slog = Logging::Logger::create(loggerJSONBackend);
+    if (g_slog == nullptr) {
+      cerr << "JSON logging to requested but it is not available" << endl;
```
Comment (fredmorcos) [inline pdns/recursordist/rec-main.cc:3049]:
```suggestion
      cerr << "JSON logging requested but it is not available" << endl;
```

File: pdns/recursordist/settings/table.py [hunk 4]
```diff
@@ -2670,8 +2670,12 @@
 - ``default``: use the traditional logging system to output structured logging information.
 - ``systemd-journal``: use systemd-journal.
   When using this backend, provide ``-o verbose`` or simular output option to ``journalctl`` to view the full information.
+- ``json``: JSON objects are written to the standard error stream
```
Comment (fredmorcos) [inline pdns/recursordist/settings/table.py:2673]:
```suggestion
- ``json``: JSON objects are written to the standard error stream.
```

**Review by fredmorcos — APPROVED** [review]

File: pdns/recursordist/docs/appendices/structuredlogging.rst (hunk 1 above)
Comment (omoerbeek) [reply pdns/recursordist/docs/appendices/structuredlogging.rst:107]: The double colon has meaning in .rst

Comment (omoerbeek) [inline pdns/recursordist/docs/appendices/structuredlogging.rst:107]:
```suggestion
An example of a a log object::
```

**Pull request merged**